src/Categories/NaturalTransformation/NaturalIsomorphism/Displayed.agda

```diff
@@ -0,0 +1,159 @@
+  ; F′⇐G′ = F′⇐G′ F′≃G′ ∘′ᵥ F′⇐G′ G′≃H′
+  ; iso′ = λ X → DisplayedIso-∘′ (iso′ F′≃G′ X) (iso′ G′≃H′ X)
+  }
+  where
+    open DMorphismₚ D′
+
+_ⓘₕ′_ : ∀ {C : Category bo bℓ be} {D : Category bo′ bℓ′ be′} {E : Category bo″ bℓ″ be″}
+          {H I : Functor D E} {F G : Functor C D}
+          {C′ : Displayed C o ℓ e} {D′ : Displayed D o′ ℓ′ e′} {E′ : Displayed E o″ ℓ″ e″}
+          {H′ : DisplayedFunctor D′ E′ H} {I′ : DisplayedFunctor D′ E′ I}
+          {F′ : DisplayedFunctor C′ D′ F} {G′ : DisplayedFunctor C′ D′ G}
+          {H≃I} {F≃G}
+      → H′ ≃[ H≃I ] I′ → F′ ≃[ F≃G ] G′ → (H′ ∘F′ F′) ≃[ H≃I ⓘₕ F≃G ] (I′ ∘F′ G′)
+_ⓘₕ′_ {E′ = E′} {I′ = I′} H′≃I′ F′≃G′ = record
+  { F′⇒G′ = F′⇒G′ H′≃I′ ∘′ₕ F′⇒G′ F′≃G′
+  ; F′⇐G′ = F′⇐G′ H′≃I′ ∘′ₕ F′⇐G′ F′≃G′
+  ; iso′ = λ X′ → DisplayedIso-resp-≈[] (DisplayedIso-∘′ (iso′ H′≃I′ _) ([ I′ ]-resp-DisplayedIso (iso′ F′≃G′ X′))) Equiv′.refl′ (⇐′.commute′ H′≃I′ (⇐′.η′ F′≃G′ X′))
+  }
+  where
+    open Displayed E′
+    open DMorphismₚ E′
+
+refl′ : ∀ {F′ : DisplayedFunctor C′ D′ F} → F′ ≃[ refl ] F′
+refl′ {D′ = D′} = record
+  { F′⇒G′ = DNT.id′
+  ; F′⇐G′ = DNT.id′
+  ; iso′ = λ X′ → record
+    { isoˡ′ = Displayed.identityˡ′ D′
+    ; isoʳ′ = Displayed.identityʳ′ D′
+    }
+  }
+
+sym′ : ∀ {F′ : DisplayedFunctor C′ D′ F} {G′ : DisplayedFunctor C′ D′ G} {F≃G}
+       → F′ ≃[ F≃G ] G′ → G′ ≃[ sym F≃G ] F′
+sym′ F′≃G′ = record
+  { F′⇒G′ = F′⇐G′ F′≃G′
+  ; F′⇐G′ = F′⇒G′ F′≃G′
+  ; iso′ = λ X′ →
+    let open DMorphism.DisplayedIso (iso′ F′≃G′ X′) in record
+    { isoˡ′ = isoʳ′
+    ; isoʳ′ = isoˡ′
+    }
+  }
+
+trans′ : ∀ {F′ : DisplayedFunctor C′ D′ F} {G′ : DisplayedFunctor C′ D′ G} {H′ : DisplayedFunctor C′ D′ H} {F≃G} {G≃H}
+         → F′ ≃[ F≃G ] G′ → G′ ≃[ G≃H ] H′ → F′ ≃[ trans F≃G G≃H ] H′
+trans′ F′≃G′ G′≃H′ = G′≃H′ ⓘᵥ′ F′≃G′
+
+isDisplayedEquivalence : {C′ : Displayed C o ℓ e} {D′ : Displayed D o′ ℓ′ e′}
+                       → IsDisplayedEquivalence isEquivalence (_≃[_]_ {C′ = C′} {D′ = D′})
+isDisplayedEquivalence = record { refl′ = refl′ ; sym′ = sym′ ; trans′ = trans′ }
+
+module LeftRightId′ (F′ : DisplayedFunctor C′ D′ F) where
+  open LeftRightId F
+  module D′ = Displayed D′
+
+  iso-id′-id′ : ∀ {X} (X′ : Displayed.Obj[_] C′ X) → DMorphism.DisplayedIso D′ {X′ = DisplayedFunctor.F₀′ F′ X′} D′.id′ D′.id′ (iso-id-id X)
+  iso-id′-id′ X′ = record { isoˡ′ = D′.identityˡ′ ; isoʳ′ = D′.identityʳ′ }
+
+-- unitors
+module _ {F : Functor C D} {F′ : DisplayedFunctor C′ D′ F} where
+  open LeftRightId′ F′
+  
+  unitorˡ′ : idF′ ∘F′ F′ ≃[ unitorˡ ] F′
+  unitorˡ′ = record { F′⇒G′ = id′∘F′⇒F′ ; F′⇐G′ = F′⇒id′∘F′ ; iso′ = iso-id′-id′ }
+
+  unitorʳ′ : F′ ∘F′ idF′ ≃[ unitorʳ ] F′
+  unitorʳ′ = record { F′⇒G′ = F′∘id′⇒F′ ; F′⇐G′ = F′⇒F′∘id′ ; iso′ = iso-id′-id′ }
+
+unitor²′ : ∀ {C′ : Displayed C o ℓ e} → idF′ ∘F′ idF′ ≃[ unitor² ] idF′ {C = C′}
+unitor²′ = record { F′⇒G′ = id′∘id′⇒id′ ; F′⇐G′ = id′⇒id′∘id′ ; iso′ = LeftRightId′.iso-id′-id′ idF′ }
+
+-- associator
+module _
```

Comment:
I've beaten my own record here, this module has 24 different `Level`s as arguments.

Reply:
Impressive indeed. And a little crazy.

Reply:
I suspect in most use cases, until we come up with a notion of "category where the objects can vary in level", this will collapse down to six (three for the base categories and three more for the displayed categories)